### DISCUSSION TITLE
Restore session COMPLETELY (Windows, unsaved changes, unsaved buffers) after normal exit or crash or whatever

### DISCUSSION THREAD

**Discussion by mickoz** [atom/atom#22312] [General]
TL;DR: Why can't we have a working session restore when we quit or after a crash (like Sublime) that will restore everything like ti was: ALL windows/project, ALL unsaved changes to already saved files, and ALL unsaved buffers?

----- Long version with more emotions and speech, etc. -----

I used Sublime.  It was perfect for my needs.  Having too much windows and tabs open, for some reason it had issues.  Then I used Atom as a backup.

Big reason Sublime was my goto text editor and notepad replacement is that I take a lot of notes.  It is fast, and it recover ALL my unsaved buffers on crash an intuitive way: you open it, everything is recovered, voilà.

Sublime crash, you reopen, everything is recovered (modifications on already saved files and unsaved buffer), voilà.

Still, I liked the idea of an open source alternative to Sublime.  Competition is not a bad thing.

However, after a surprise electricity outage this morning (and me procrastinating saving my text buffer AND not buying an UPS of course), guest what, lost again my unsaved buffers in Atom.  I even found a snippet sometime ago to recover unsaved buffer, but to no success, some unsaved buffer were not found back with that code.  Lost recent notes I did not want to lose.

Either that is enough for me unfortunately (that feature is a deal breaker as a notepad replacement), either someone take this seriously, or either someone tell me how to have the same behavior (maybe I am just dumb and it exists already, but why can't this work out of the box without me having to do hours of google search?).

I don't like to whine, but I also like a lot improvement.  I guest Steve Job would say: that is not good enough, go back to drawing board until it does it smoothly.  With this post, I hope it can trigger something.  I have found 3 gazillions post about it.  Maybe it is not as important for everyone as for me, but I am sure not alone.

I think ANY program nowadays should be tolerant to crash, etc.  Text editor, browser, picture editor, spreadsheet, etc.  All program.  It should not be that hard to support a kind of session restore.

My software crashing, I don't like of course, but losing my data = end of the world. ;-)

I see that there are those options in Atom, I enabled them, it doesn't work.  Inner self scream: WHY THIS DOESN'T JUST WORK?  IS IT ME?

I want to be able to quit and reopen and everything restored (project concept or not, I don't care!).

I want everything to be able to crash and when I reopen it does the same as if I quit and reopened (I can understand if this might have a delay for technical/performance reason, e.g., 30 seconds, 1 mins, etc.).  But leaving my computer for hours, computer crashing, should be able to restart as it.  Especially for something like taking notes, writing important text, etc. (at least that is my use for such editor, even if I understand lot of people, it is their coding stuff, but I want the same for everything, my code editor too which can be Sublime or Atom depending on the hack or project I work on, eh).

Is that not important enough?

All of us experience crash and are vulnerable to data crash.

**Comment by lenora2468B** [comment]
See- this has happened like once a week for me and somehow I do enable it but I agree 100 percent that it should just work again. “Reboot”

**Comment by Bofahda** [comment]
I experienced this problem recently.  Atom always opened on the tabs I was working on, but now it opens on a different project.  My old project does not show under "repoen last project".  Is there a solution to find my lost tabs or do I need to start all over from the beginning?
